src/cli/serve/compile_config_stack.js

```diff
@@ -64,6 +77,21 @@ function getServerlessModeFromCfg(configs) {
   return config.serverless;
 }
 
+/** @typedef {'search_ai_lake' | 'essentials' | 'complete'} ServerlessSecurityTier */
+/**
+ * @param {string[]} configs List of configuration file paths
+ * @returns {ServerlessSecurityTier|undefined} The serverless security tier in the summed configs
+ */
+function getSecurityTierFromCfg(configs) {
+  const config = getConfigFromFiles(configs.filter(isNotNull));
+
+  const productType = _.get(config, 'xpack.securitySolutionServerless.productTypes', [])[0];
```

Comment:
The logic assumes the first `product_tier` is representative. There's no validation, but it should always be the same in all the product types in the array.